src/objprint/objprint.py

```diff
@@ -279,6 +285,46 @@ def add_indent(
             return " " * (indent_level * cfg.indent) + line
         return [" " * (indent_level * cfg.indent) + ll for ll in line]
 
+    def register_formatter(
+        self,
+        obj_type: Type[Any],
+        obj_formatter: Optional[Callable[[Any], str]] = None
+    ) -> Optional[Callable[[Callable[[Any], str]], Callable[[Any], str]]]:
+        if obj_formatter is None:
+            def wrapper(obj_formatter: Callable[[Any], str]) -> Callable[[Any], str]:
+                self.register_formatter(obj_type, obj_formatter)
+                return obj_formatter
+            return wrapper
+        self._validate_formatter(obj_type, obj_formatter)
+        self.type_formatter[obj_type] = obj_formatter
+        return None
+
+    def unregister_formatter(self, *obj_types: Type[Any]) -> None:
+        if not obj_types:
+            self.type_formatter.clear()
+        else:
+            for obj_type in obj_types:
+                if obj_type in self.type_formatter:
+                    del self.type_formatter[obj_type]
+
+    def list_formatter(self, print_list: bool = True) -> Optional[dict]:
```

Comment:
Having a function to list all the formatters is good, but returning a formatted string is not a great idea. A common use would be to check whether a type already has a formatter. Just return the dictionary would be fine. It's a well-defined format and the user can do whatever they want with the dict.

Reply:
What the function returns is the dictionary. It is printing the formatted string because the type stored is something like `<class 'int'>`, and I thought it would be more clear to just print `int`. I think it make sense to use `self._objstr()` to get the string and print it out.

Reply:
Don't print things if not explicitly issued from the users. I'd suggest to change it to `get_formatters()` and just return a dict. Why would it return None? The user can print with a specific format if they want, but printing the dict itself is clear enough for simple usages. It's not `objprint`'s duty to format this formatter dict well.